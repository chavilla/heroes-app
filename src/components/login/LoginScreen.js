import React, {
  useContext
} from "react";
import {
  AuthContext
} from "../auth/AuthContext";
import {
  types
} from "../types/types";

export const LoginScreen = ({
  history
}) => {
  const {
    dispatch
  } = useContext(AuthContext);

  const handleLogin = () => {

    const lastPath = localStorage.getItem("lastPath") || "/";

    dispatch({
      type: types.login,
      payload: {
        name: "Jes√∫s",
      },
    });

    history.replace(lastPath);
  };

  return ( <
    div className = "container pt-5" >
    <
    h1 className = "text-center" > Login < /h1> <
    hr / >
    <
    button className = "btn btn-primary w-25 m-auto"
    onClick = {
      handleLogin
    } >
    Ingresar <
    /button> <
    /div>
  );
};