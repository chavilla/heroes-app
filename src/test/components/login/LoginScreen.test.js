import React from "react";
import { mount } from "enzyme";
import { LoginScreen } from "../../../components/login/LoginScreen";
import { AuthContext } from "../../../components/auth/AuthContext";
import { types } from "../../../components/types/types";

describe("<LoginScreen />", () => {
  const history = {
    replace: jest.fn(),
  };

  const context = {
    dispatch: jest.fn(),
  };

  const wrapper = mount(
    <AuthContext.Provider value={context}>
      <LoginScreen history={history} />
    </AuthContext.Provider>
  );

  test("should to display the component", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("should to call history replace", () => {
    const handleLogin = wrapper.find("button").prop("onClick");
    handleLogin();
    expect(history.replace).toHaveBeenCalledWith("/");
    expect(context.dispatch).toHaveBeenCalledWith({
      type: types.login,
      payload: {
        name: "Jes√∫s",
      },
    });

    localStorage.setItem("lastPath", "/dc");

    handleLogin();

    expect(history.replace).toHaveBeenCalledWith("/dc");
  });
});
