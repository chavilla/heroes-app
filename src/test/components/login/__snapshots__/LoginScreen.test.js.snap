// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoginScreen /> should to display the component 1`] = `
<div
  className="container pt-5"
>
  <h1
    className="text-center"
  >
     Login 
  </h1>
   
  <hr />
  <button
    className="btn btn-primary w-25 m-auto"
    onClick={[Function]}
  >
    Ingresar 
  </button>
   
</div>
`;
